tests/stdlib/trstgen.nim

```diff
@@ -1144,10 +1145,33 @@ Test1
       """
     var error = new string
     let output = input.toHtml(error=error)
-    check(error[] == "input(1, 1) Error: invalid field: " &
+    check(error[] == "input(2, 3) Error: invalid field: " &
                      "extra arguments were given to number-lines: ' let a = 1'")
     check "" == output
 
+  test "code-block warning":
```

Comment:
not a blocker because the situation after this PR improves, but the line numbers and columns are off by ~1 for nested doc comments, eg:

before PR: all lines/cols are 1 (useless)
after PR: main3 is correct, main1 and main2 are off by ~1

```nim

proc main1*()=
  ##[
    .. code:: Nim
       :unsupportedField: anything
    .. code:: unsupportedLang
       anything
    ```anotherLang
    someCode
  ```
  ]##
  discard


proc main2*()=
  ##[
  .. code:: Nim
     :unsupportedField: anything
  .. code:: unsupportedLang
     anything
  ```anotherLang
  someCode
  ```
  ]##
  discard

proc main3*()=
  ## .. code:: Nim
  ##    :unsupportedField: anything
  ## .. code:: unsupportedLang
  ##    anything
  ## ```anotherLang
  ## someCode
  ## ```
  discard
```

Reply:
yes, line info is saved only for first `#` of `##`, not for real comment inside.
We talked about this problem in https://github.com/timotheecour/Nim/issues/658